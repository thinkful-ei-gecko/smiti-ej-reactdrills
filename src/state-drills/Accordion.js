import React, {Component} from 'react';
import cuid from 'cuid';

class Accordion extends Component{

    static defaultProps = {
        sections: [],
    };

    state = {
        currentSection: 0
    };

    handleButtonClicked(index) {
        this.setState({
            currentSection: index
        })
    }

    render(){
        console.log(this.state.currentSection);
        const buttons = this.props.sections.map((section, index) => {

        let id = cuid()

        if (this.state.currentSection === index) {
            return (
            <li key={id}>
                <button onClick={() => this.handleButtonClicked(index)}>{section.title}</button>
                <p>{section.content}</p>
            </li>
            );
        }

        return (
            <li key={id}>
                <button onClick={() => this.handleButtonClicked(index)}>{section.title}</button>
            </li>
        )})

        return (
            <ul>
            {buttons}
            </ul>
        )
    }


}

export default Accordion;