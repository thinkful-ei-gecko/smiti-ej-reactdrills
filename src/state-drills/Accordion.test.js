import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';
import Sections from './Sections';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({
    adapter: new Adapter()
});

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion  sections={[{}]} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

it('renders with empty li', () => {
    let wrapper = shallow(<Accordion />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it('renders no sections as active by default', () => {
    let wrapper = shallow(<Accordion sections={Sections} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it('the component opens a clicked section', () => {
    let wrapper = shallow(<Accordion sections={Sections} />);
    wrapper.find('button').at(0).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot();
});

it('opens the last section when multiple sectionns have been clicked', () => {
    let wrapper = shallow(<Accordion sections={Sections} />);
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot();
});