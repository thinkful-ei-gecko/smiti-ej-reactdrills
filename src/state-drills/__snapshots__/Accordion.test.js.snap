// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`opens the last section when multiple sectionns have been clicked 1`] = `
<ul>
  <li
    key="ck077o8h0000gvzi3g2kg3nj8"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="ck077o8h0000hvzi35rpb5mik"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="ck077o8h0000ivzi369ic0un8"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`renders no sections as active by default 1`] = `
<ul>
  <li
    key="ck077o8gb0001vzi3apdz8bmh"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    key="ck077o8gb0002vzi31z0fexbt"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="ck077o8gb0003vzi3ckn947ek"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`renders with empty li 1`] = `<ul />`;

exports[`renders without crashing 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`the component opens a clicked section 1`] = `
<ul>
  <li
    key="ck077o8go0007vzi34zbv59bo"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    key="ck077o8go0008vzi3hss88ktr"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="ck077o8go0009vzi34p7wfzbr"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
